src/frontend/src/expr/function_call.rs

```diff
@@ -87,11 +87,29 @@ impl std::fmt::Debug for FunctionCall {
 impl FunctionCall {
     /// Create a `FunctionCall` expr with the return type inferred from `func_type` and types of
     /// `inputs`.
-    pub fn new(func_type: ExprType, inputs: Vec<ExprImpl>) -> Result<Self> {
-        let return_type = infer_type(
-            func_type,
-            inputs.iter().map(|expr| expr.return_type()).collect(),
-        )?; // should be derived from inputs
+    pub fn new(func_type: ExprType, mut inputs: Vec<ExprImpl>) -> Result<Self> {
+        let return_type = match func_type {
+            ExprType::Case => {
+                let len = inputs.len();
+                align_types(inputs.iter_mut().enumerate().filter_map(|(i, e)| {
+                    // `Case` organize `inputs` as (cond, res) pairs with a possible `else` res at
+                    // the end. So we align exprs at odd indices as well as the last one when length
+                    // is odd.
+                    match (i & 1) == 1 || (len & 1) == 1 && i == len - 1 {
```

Comment:
```suggestion
                    match i.is_odd() || len.is_odd() && i == len - 1 {
```

Reply:
Thanks! I did not find this in std or num-traits ...